src/internals/StreamHttpRequest.hpp

```diff
@@ -370,6 +379,41 @@ const ArduinoHttpServer::ErrorString ArduinoHttpServer::StreamHttpRequest<MAX_BO
    return errorString;
 }
 
+template <size_t MAX_BODY_SIZE>
+bool ArduinoHttpServer::StreamHttpRequest<MAX_BODY_SIZE>::authenticate(const char * username, const char * password) const
+{
+   if (m_authorizationField.getType() == HttpField::Type::NOT_SUPPORTED)
+   {
+      return false;
+   }
+
+   if (!m_authorizationField.getValueAsString().startsWith("Basic"))
+   {
+      DEBUG_ARDUINO_HTTP_SERVER_PRINTLN("Unsupported auth header");
+      return false;
+   }
+   String authData = m_authorizationField.getValueAsString().substring(6);
```

Comment:
Any reason not to directly use `FixString`? `m_authorizationField.substring(6)`. I would like to restrain from using String to prevent dynamic allocation. You could also assign the result to `FixSTring<some size>` authData. The reserve would then also be unnecessary.

Reply:
I just didnt know what max length to choose as the HTTP standard does not define one AFAIK. If I choose something too small than it will limit the combined length of user+password.
Let me know if you prefer to define static limit or to keep the dynamic one. I'm fine both ways.

Reply:
Oké, let's decide to keep this as is for now.

Reply:
I will fix the substring(6) one. That will be easy.